metricbeat/module/kafka/broker.go

```diff
@@ -91,6 +98,11 @@ func NewBroker(host string, settings BrokerSettings) *Broker {
 		cfg.Net.SASL.Enable = true
 		cfg.Net.SASL.User = user
 		cfg.Net.SASL.Password = settings.Password
+		err := settings.Sasl.ConfigureSarama(cfg)
+
+		if err != nil {
+			return nil
+		}
```

Comment:
I added a `Validate()` func to the struct. Though the `ConfigureSarama` still has the `switch` with `default`. Should I just remove that and also not check for `err` in `broker.go`? Then I could also change the signature to remove the error return value altogether.

Reply:
Yes, I think this could be a good refactor. Maybe we can keep the `default` case with a panic, as a case that should never happen, but just in case, to detect it earlier if happens.